### PR TITLE
Call-Control resource

### PR DESCRIPTION
**Pull request by luizanao**
### Call Control

This PR introduces ability do hook call-control related actions using telnyx-python SDK

#### Include:
- answer
- fork_start
- fork_stop
- gather_using_audio
- gather_using_speak
- playback_start
- playback_stop
- record_start
- record_stop
- send_dtmf
- speak
- transfer
- reject
- hangup
- bridge

#### Usage: 
```python
call = telnyx.CallControl.create(
		connection_id="1148443925661353289",
		to="+13122010002",
		from_="+13122010001"
	)
call.call_control_id = '123' #call_control_id is received by webhook callback
call.answer()
```
**tests depends on** https://github.com/team-telnyx/telnyx-mock/pull/1

### CODE REVIEW

File: telnyx/api_resources/abstract/nested_resource_class_methods.py [hunk 1]
```diff
@@ -4,17 +4,27 @@
 from telnyx.six.moves.urllib.parse import quote_plus
 
 
-def nested_resource_class_methods(resource, path=None, operations=None):
+def nested_resource_class_methods(
+    resource, path=None, operations=None, quote_params=True
```
Comment (nicktimko) [inline telnyx/api_resources/abstract/nested_resource_class_methods.py]: The parameter is called `quote_params` but I don't see how you're passing URL parameters. Parameters as in the function, not any argument? How about `quote_path_parts`?

Is this a workaround for call control not using URL-safe IDs?

Comment (luizanao) [reply telnyx/api_resources/abstract/nested_resource_class_methods.py]: Thanks for the comment @nicktimko .
I agree that this parameter name is kinda confuse. `quote_path_parts` would be much more clear.
And yes, this is actually for both. 
For call-control we haven't URL safe and for `path` has been always escaped 
e.g `actions/extend` becomes `actions%2Fextend`

Comment (luizanao) [reply telnyx/api_resources/abstract/nested_resource_class_methods.py]: @nicktimko @IgnasiBosch I sent another patch based on this comment. Thanks again for that, very valuable.
My solution: We're using quote_plus that is pretty much a proxy method to quote() (with some logic), but overriding  `safe` func param.
see: [cpyton/urllib/parse.py](https://github.com/python/cpython/blob/eed5e9a9562d4dcd137e9f0fc7157bc3373c98cc/Lib/urllib/parse.py#L787)

Now, we're just using builtin quote_plus safe param to ensure paths will always be escaped. Also, we have a "list" of safe chars we want to use in the path `id` param.  (such =,- )

File: telnyx/util.py [hunk 2]
```diff
@@ -18,6 +18,9 @@
 
 __all__ = ["io", "parse_qsl", "utf8", "log_info", "log_debug", "logfmt"]
 
+# Special valid characters in telnyx id's objects. Format: `=/$#`
+telnyx_valid_id_parts = "=-"
```
Comment (nicktimko) [inline telnyx/util.py]: `-` should always be OK, unescaped or no
`=` "should" always be escaped (according to [RFC-3986](https://tools.ietf.org/html/rfc3986)) unless it's actually part of a query string, but servers generally seem to be OK with it. I'm not sure if in this SDK it'd be escaped twice, once that we bypass using this, and again by the actual library (requests, urlfetch, etc.).

**Pull request merged**